openedx/core/djangoapps/content_staging/serializers.py

```diff
@@ -35,6 +45,29 @@ class UserClipboardSerializer(serializers.Serializer):
     source_usage_key = serializers.CharField(allow_blank=True)
     # The title of the course that the content came from originally, if relevant
     source_context_title = serializers.CharField(allow_blank=True, source="get_source_context_title")
+    # The URL where the original content can be seen, if it still exists and the current user can view it
+    source_edit_url = serializers.SerializerMethodField(source="get_source_edit_url")
+
+    def get_source_edit_url(self, obj) -> str:
+        """ Get the URL where the user can edit the given XBlock, if it exists """
+        request = self.context.get("request", None)
+        user = request.user if request else None
+        if not user:
+            return ""
+        if not obj.source_usage_key.context_key.is_course:
+            return ""  # Linking back to libraries is not implemented yet
+        if not has_studio_read_access(user, obj.source_usage_key.course_key):
+            return ""
+        try:
+            block = modulestore().get_item(obj.source_usage_key)
+        except ItemNotFoundError:
+            return ""
+        edit_url = xblock_studio_url(block, find_parent=True)
+        if not edit_url:
+            return ""
+        if request:
```

Comment:
This is always true. Otherwise, `user` would be `None`, and this method would return on line 56.